### ISSUE TITLE
Discovery: redefine drop zones in the canvas

### ISSUE THREAD

**Issue by dvoytenko** [GoogleForCreators/web-stories-wp#145]
Per recent discussion, the drop zones in the canvas will be mainly focused on the:

1. DND from media panel to the editor
2. DND from the editor to editor (extension to the selection Moveable)

As a stretch goal, some adjustments to DND from desktop to editor (#54). However, that's not the highest priority for this issue.

Some key features:

1. DND from the medial panel should give a good preview of what adding an element to the page would be like. E.g. ideally it'd use the real scaled element dimensions, not just thumbnail. The dimensions should probably be similar to the ones assigned to an element when user simply clicks on an item in the media panel.
2. Drop zones will likely be defined either as borders (background, masked shapes, layouts in the future).
3. "Snapping into" and "snapping out" behavior should be defined.
4. "Snapping" preview behavior should be defined.

**Comment by spacedmonkey** [comment]
Blocker for #229 #228

**Comment by dvoytenko** [comment]
@spacedmonkey, this issue is designed to be explicitly not conflicting or blocking #229 and #228. Where do you see blockages?

**Comment by spacedmonkey** [comment]
I have already made a start on #228 I had a thought. When I file is dragged into the window, the drop zone would become visual. Highlighting when a file can dropped on screen and what happens when it is dropped. Until this behavior is confirmed / defined, it makes me feel we should wait on merging #228 . Thoughts?

**Comment by dvoytenko** [comment]
@spacedmonkey I think we need to definitely clarify how #228 works an external resource is dropped on the workspace area, design panel, etc. But it should be fully independent of this issue.

**Comment by pbakaus** [comment]
I've completed this here: https://docs.google.com/document/d/1_3-_v4zH6bW9hOplOO8wYxj8l4C0soMgEbD7XLiXrrg/edit#heading=h.dd0961bhpsrd (let me know if you need any more detail).
